torch/csrc/dynamo/compiled_autograd_py.cpp

```diff
@@ -0,0 +1,484 @@
+#include <torch/csrc/dynamo/compiled_autograd_py.h>
+
+#include <torch/csrc/autograd/compiled_autograd.h>
+#include <torch/csrc/autograd/engine.h>
+#include <torch/csrc/jit/python/pybind_utils.h>
+#include <torch/csrc/python_headers.h>
+#include <torch/csrc/utils/pythoncapi_compat.h>
+#include <iostream>
+#include <vector>
+
+namespace torch {
+namespace dynamo {
+using namespace torch::autograd;
+using c10::SymInt;
+
+static PyObject* wrap_variable_list(const variable_list& inputs) {
+  PyObject* pyinput = PyList_New(inputs.size());
+  for (const auto i : c10::irange(inputs.size())) {
+    PyList_SET_ITEM(pyinput, i, THPVariable_Wrap(inputs[i]));
+  }
+  return pyinput;
+}
+
+static PyObject* wrap_int_list(const std::vector<int64_t>& inputs) {
+  PyObject* pyinput = PyTuple_New(inputs.size());
+  for (const auto i : c10::irange(inputs.size())) {
+    PyTuple_SET_ITEM(pyinput, i, PyLong_FromSsize_t(inputs[i]));
+  }
+  return pyinput;
+}
```

Comment:
Consider

```
template <typename T, typename WrapFn>
PyObject* wrap_list(const T& inputs, WrapFn&& wrapFn) {
  PyObject* pyinput = PyTuple_New(inputs.size());
  for (const auto i : c10::irange(inputs.size())) {
    PyTuple_SET_ITEM(pyinput, i, wrapFn(inputs[i]));
  }
  return pyinput;
}

PyObject* wrap_variable(const Variable& var) {
  return THPVariable_Wrap(var);
}

PyObject* wrap_int(int64_t val) {
  return PyLong_FromSsize_t(val);
}
```
No strong opinions :)

Reply:
One is a list and one is a tuple (and the python code requires that iirc)